src/resolvers/journalResolver.js

```diff
@@ -12,26 +13,63 @@ const journalResolver = {
   },
 
   Mutation: {
-    createJournal: async (_, args) => {
-      const { title, url, issn, domainName, policies } = args.journal;
-      const journal = new Journal({ title, url, issn, domainName, policies });
-      await journal.save();
-      return journal;
+    createJournal: async (_, { journalToCreate }, { req }) => {
+      try {
+        const currentUser = await User.findById(req.session.userId);
+
+        const journal = new Journal({
+          createdBy: req.session.userId,
+          ...journalToCreate,
+        });
+
+        const createdJournal = await journal.save();
+
+        currentUser.journals.push(createdJournal.id);
+
+        await currentUser.save();
+
+        return { journal };
+      } catch (error) {
+        if (error.code === 11000 && Object.keys(error.keyValue)[0] === "issn") {
+          return {
+            errors: [{ field: "issn", message: "issn already exists" }],
```

Comment:
Change the error message to:
```
A journal with the same ISSN already exists
```

It would be even better if we use the string template literal & display the ISSN in the error message